web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,360 @@
+        user['total_additions'] = 0;
+        user['total_deletions'] = 0;
+        user['max_contribution_type'] = 0;
+        const filteredWeeks = user.weeks.filter((week) => {
+          const weekDate = new Date(week.week);
+          if (weekDate >= this.dateFrom && weekDate <= this.dateUntil) {
+            user['total_commits'] += week.commits;
+            user['total_additions'] += week.additions;
+            user['total_deletions'] += week.deletions;
+            if (week[this.type] > user['max_contribution_type']) {
+              user['max_contribution_type'] = week[this.type];
+            }
+            return true;
+          }
+          return false;
+        });
+
+        filteredData[key] = {...user, weeks: filteredWeeks};
+      }
+
+      return filteredData;
+    },
+
+    maxMainGraph() {
+      const maxValue = Math.max(
+        ...this.totalStats.weeks.map((o) => o[this.type])
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
+    maxContributorGraph() {
+      const maxValue = Math.max(
+        ...this.sortedContributors.map((c) => c['max_contribution_type'])
+      );
+      const [cooefficient, exp] = maxValue
+        .toExponential()
+        .split('e')
+        .map(Number);
+      if (cooefficient % 1 === 0) {
+        return maxValue;
+      }
+      return (1 - (cooefficient % 1)) * 10 ** exp + maxValue;
+    },
+
+    toGraphData(data) {
+      const style = getComputedStyle(document.body);
+      const colorName = this.type === 'commits' ? '--color-primary-alpha-60' : (this.type === 'additions' ? '--color-green-badge-hover-bg' : '--color-red-badge-hover-bg');
+      const color = style.getPropertyValue(colorName).trim();
+      return {
+        datasets: [
+          {
+            data: data.map((i) => {
+              return {x: i.week, y: i[this.type]};
+            }),
+            pointRadius: 0,
+            pointHitRadius: 0,
+            fill: 'start',
+            backgroundColor: color,
+            borderWidth: 0,
+            tension: 0.3,
+          },
+        ],
+      };
+    },
+
+    updateOtherCharts(event, reset) {
+      const minVal = event.chart.options.scales.x.min;
+      const maxVal = event.chart.options.scales.x.max;
+      if (reset) {
+        this.dateFrom = this.startDate;
+        this.dateUntil = this.endDate;
+        this.sortContributors();
+      } else if (minVal) {
+        this.dateFrom = new Date(minVal);
+        this.dateUntil = new Date(maxVal);
+        this.sortContributors();
+      }
+    },
+
+    getOptions(type) {
+      return {
+        responsive: true,
+        maintainAspectRatio: false,
+        animation: false,
+        onClick: (e) => {
+          if (type === 'main') {
+            e.chart.resetZoom();
```

Comment:
If I make it work with double click, I will probably need to inform the user about it otherwise it won't be noticed.

I think there is no harm if it stays like this. You can always zoom again, it shouldn't be a big deal :)

Reply:
@delvh What do you think about this now? Should I left it as is? Did you get a chance to test it?